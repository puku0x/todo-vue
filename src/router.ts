import { createRouter, createWebHistory, RouteRecordRaw } from 'vue-router';

const routes: Array<RouteRecordRaw> = [
  { path: '/', redirect: '/todos' },
  {
    path: '/todos',
    component: () => import('./pages/todo/todo-list').then(m => m.TodoListPage)
  },
  {
    path: '/todos/new',
    component: () =>
      import('./pages/todo/todo-create').then(m => m.TodoCreatePage)
  },
  {
    path: '/todos/:id',
    component: () =>
      import('./pages/todo/todo-detail').then(m => m.TodoDetailPage)
  },
  {
    path: '/todos/:id/edit',
    component: () => import('./pages/todo/todo-edit').then(m => m.TodoEditPage)
  }
];

export const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes
});
