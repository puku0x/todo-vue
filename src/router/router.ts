import { createRouter, createWebHistory, RouteRecordRaw } from 'vue-router';

const routes: Array<RouteRecordRaw> = [
  { path: '/', redirect: '/todos' },
  {
    path: '/todos',
    component: () => import('../pages/todo').then(m => m.TodoListPage)
  }
];

export const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes
});
