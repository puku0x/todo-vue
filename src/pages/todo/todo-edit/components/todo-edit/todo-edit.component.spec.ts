import { mount } from '@vue/test-utils';
import { createRouter, createMemoryHistory } from 'vue-router';

import { Todo } from '@/models';
import { routes } from '@/router';

import TodoEdit from './todo-edit.component.vue';

describe('TodoEdit', () => {
  it('render', async () => {
    const todo: Todo = {
      id: '1',
      title: 'title',
      completed: false,
      createdAt: 123456789,
      updatedAt: 123456789,
    };
    const router = createRouter({
      history: createMemoryHistory(),
      routes,
    });
    router.push('/todos/1/edit');
    await router.isReady();
    const wrapper = mount(TodoEdit, {
      global: {
        plugins: [router],
      },
      props: {
        isFetching: false,
        todo,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
