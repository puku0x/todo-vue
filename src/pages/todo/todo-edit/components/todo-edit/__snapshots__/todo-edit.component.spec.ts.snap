// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoEdit render 1`] = `
<a href="/todos" class="">Back to list</a>
<h2>todo-edit</h2>
<form novalidate="">
  <p><button type="submit">Save</button></p>
  <table>
    <tbody>
      <tr>
        <td>title</td>
        <td><input type="text" name="title"></td>
      </tr>
      <tr>
        <td>completed</td>
        <td><input type="checkbox" name="completed"></td>
      </tr>
    </tbody>
  </table>
</form>
`;
